code/modules/mob/language/animal.dm

```diff
@@ -0,0 +1,32 @@
+/decl/language/animal
+	name = "Animal Noises" // translate them!
+	desc = "Some varieties of animal can communicate amongst themselves, apparently."
+	colour = "say_quote"
+	key = "a"
+	shorthand = "A"
+	hidden_from_codex = TRUE
+
+/decl/language/animal/can_be_understood_by(var/mob/living/speaker, var/mob/living/listener)
+	if(!istype(listener))
+		return FALSE
+	if(listener.universal_understand || listener.universal_speak)
+		return TRUE
+	if(!istype(speaker) || speaker.universal_speak)
+		return TRUE
```

Comment:
Confusing and probably illegal check; if `speaker` is mistyped, you shouldn't be checking vars on it.

Reply:
I was wrong here, sorry.